docs/core/diagnostics/diagnostic-libraries.md

```diff
@@ -0,0 +1,161 @@
+
+Extending this example, you can leverage [Spectre.Console](https://www.nuget.org/packages/Spectre.Console), a well-regarded .NET library designed to simplify the development of visually appealing, cross-platform console applications. With Spectre, you'll be able to present resource utilization data in a tabular format. The following code illustrates the usage of the `IResourceMonitor` interface to access details regarding the CPU and memory usage of the current process, then presenting this data in a table:
+
+:::code source="snippets/resource-monitoring/Program.cs" id="monitor" highlight="30-31,36-40":::
+
+The preceding code:
+
+- Creates a cancellation token source and a cancellation token.
+- Creates a new `Table` instance, configuring it with a title, caption, and columns.
+- Performs a live render of the `Table` instance, passing in a delegate that will be invoked every three seconds.
+- Gets the current resource utilization information from the `IResourceMonitor` instance and displays it as a new row in the `Table` instance.
+
+The following is an example of the output from the preceding code:
+
+:::image type="content" source="media/resource-monitoring-output.png" lightbox="media/resource-monitoring-output.png" alt-text="Example Resource Monitoring app output.":::
+
+## Health checks
+
+In a distributed system, health checks are periodic assessments of the status, availability, and performance of individual nodes or services. These checks ensure that the system functions correctly and efficiently. Health checks are essential for system reliability, and they are typically performed at regular intervals with the results analyzed for decision-making and corrective actions.
+
+The following heath check status results are possible:
+
+- <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus.Healthy?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus.Degraded?displayProperty=nameWithType>
+- <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus.Unhealthy?displayProperty=nameWithType>
+
+### Application lifetime health checks
+
+To perform health checks on the application lifetime events of <xref:Microsoft.Extensions.Hosting.IHostApplicationLifetime>, use the <xref:Microsoft.Extensions.Diagnostics.HealthChecks.CommonHealthChecksExtensions.AddApplicationLifecycleHealthCheck%2A> extension method available in the [Microsoft.Extensions.Diagnostics.HealthChecks.Common](https://www.nuget.org/packages/Microsoft.Extensions.Diagnostics.HealthChecks.Common) NuGet package.
+
+This provider will indicate that the application is healthy only when it is fully active. Until the lifetime object indicates the application has started, the provider will report the application as not healthy. When the application starts shutting down, the provider will report the application as unhealthy.
```

Comment:
```suggestion
This provider will signal that the application is healthy when lifetime object will indicate the application has started. When the application starts shutting down, the provider will report the application as unhealthy.
```

I've seen a repetition of 'indicates' word twice, and during the process of replacing it, I did more changes that shortens the sentence without losing information.